Board.java

```diff
@@ -13,15 +13,29 @@ public String getState() {
         return this.state;
     }
 
+     public Boolean avaliablePos(int pos) {
+        if (state.charAt(pos) == '-') {
+            if (!(pos > 9) || !(pos < 1)) {
```

Comment:
Is this inner loop even necessary? Also, this conditional is complicated. A simpler way would be pos >= 0 && pos < board.length